Bump gitpython to address CVE

chmeliik opened the pull request:
# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- N/A Code coverage from testing does not decrease and new code is covered
- N/A New code has type annotations
- N/A Docs updated (if applicable)
- N/A Docs links in the code are still valid (if docs were updated)

taylormadore reviewed (approved):
nit: typo in the commit message that I didn't catch on the cachito PR

chmeliik commented:
> nit: typo in the commit message that I didn't catch on the cachito PR

fixed and removed duplicate word ("unsafe _again_ in a requirements file _again_")
